### ISSUE TITLE
Eclipse Oxygen.3a, RED 0.8.6.201806181254, pydevd-1.2.0 and runPyDevDebug.py

### ISSUE THREAD

**Issue by saskell** [nokia/RED#227]
I had a heck of a time following the instructions in http://nokia.github.io/RED/help/user_guide/launching/debug/robot_python_debug.html and getting python debugging to work.

I would get this error: "-m" is not recognized as an internal or external command.  See below for details.

I finally hacked runPyDevDebug.py to where it would run - see attached.

So what am I doing wrong?

Thanks!
--saskell
[runPyDevDebug.py.txt](https://github.com/nokia/RED/files/2120913/runPyDevDebug.py.txt)

Command: D:\local\vpy\Lev3-Robot\Scripts\python.exe d:\local\tools\REDScripts\runPyDevDebug.py -m robot.run --listener C:\Users\sckell\AppData\Local\Temp\RobotTempDir7907313046246815718\TestRunnerAgent.py:52922 -P D:\local\vpy\Lev3-Robot\Lib\site-packages:C:\dev\gitlab\sckell\Lev3-Robot\common:C:\dev\gitlab\sckell\Lev3-Robot -s Lev3-Robot.RegexEventsAttrTNGAuto1659 --variable project_path:C:\dev\gitlab\sckell\Lev3-Robot C:\dev\gitlab\sckell\Lev3-Robot
Suite Executor: Robot Framework 3.0.4 (Python 3.6.4 on win32)
'"-m"' is not recognized as an internal or external command,
operable program or batch file.
Running command: "-m" -u -m pydevd --multiprocess --print-in-debugger-startup --vm_type python --client 127.0.0.1 --port 5678 --file D:/local/vpy/Lev3-Robot/lib/site-packages/robot/run.py "C:\Users\sckell\AppData\Local\Temp\RobotTempDir7907313046246815718\TestRunnerAgent.py:52922" "-P" "D:\local\vpy\Lev3-Robot\Lib\site-packages:C:\dev\gitlab\sckell\Lev3-Robot\common:C:\dev\gitlab\sckell\Lev3-Robot" "-s" "Lev3-Robot.RegexEventsAttrTNGAuto1659" "--variable" "project_path:C:\dev\gitlab\sckell\Lev3-Robot" "C:\dev\gitlab\sckell\Lev3-Robot"

**Comment by modrz** [comment]
Hi,

It looks like process run command is not created correctly.

Can you try following script:
[runPyDevDebug.py.txt](https://github.com/nokia/RED/files/2123643/runPyDevDebug.py.txt)

**Comment by saskell** [comment]
Thanks modrz!

I tried your file and it failed... it looked like the --listener option was getting munged so I changed this line in LocalPythonDebug() from:

command = [sys.executable, '-u'] + pydevdPath + pydevdArgs + sys.argv[4:]

to:

command = [sys.executable, '-u'] + pydevdPath + pydevdArgs + sys.argv[3:]

Do you agree with this change (full modified file [runPyDevDebug.py.txt](https://github.com/nokia/RED/files/2124206/runPyDevDebug.py.txt) attached)?

Thanks!
Scott

**Comment by modrz** [comment]
Actually command creation line should look like:

command = [sys.executable, '-u'] + pydevdPath + pydevdArgs + sys.argv[sys.argv.index('robot.run') + 1:]

**Comment by saskell** [comment]
Yes - that's a better more robust implementation.  And I tested it out and it works.

I think that version should be committed to the repo (?)

Thanks for your help!

--Scott

**Comment by modrz** [comment]
Ok, thanks for confirmation.
Script will be released with next version.
Moreover it will be fixed to work on linux also.

**Comment by KrzysztofJozefowicz** [comment]
released in 0.8.7
